src/software/ai/hl/stp/tactic/goalie_tactic.cpp

```diff
@@ -71,67 +150,145 @@ void GoalieTactic::calculateNextIntent(IntentCoroutine::push_type &yield)
-            // if the ball is slow or stationary inside our defense area, chip it out
-            next_intent = chip_action.updateStateAndGetNextIntent(
-                *robot, ball, ball.position(),
-                (ball.position() - field.friendlyGoal()).orientation(), 2);
+            // if the ball is slow but its not safe to chip it out, don't.
+            // TODO finesse the ball out of the goal using the dribbler.
+            // for now we just stop https://github.com/UBC-Thunderbots/Software/issues/744
+            if (dont_chip_rectangle.containsPoint(ball.position()) == true)
+            {
+                next_intent = stop_action.updateStateAndGetNextIntent(*robot, false);
+            }
+            // if the ball is slow or stationary inside our defense area, and is safe
+            // to do so, chip it out
+            else
+            {
+                next_intent = chip_action.updateStateAndGetNextIntent(
+                    *robot, ball, ball.position(),
+                    (ball.position() - field.friendlyGoal()).orientation(), 2);
+            }
         }
-        // Case 3
+        // case 3: ball does not have a clear velocity vector towards the goal, so
+        // position goalie in best position to block shot
         else
         {
-            // block the cone by default
-            float radius =
-                Util::DynamicParameters::GoalieTactic::block_cone_buffer.value() +
-                ROBOT_MAX_RADIUS_METERS;
-            Point goalie_pos =
-                calcBlockCone(field.friendlyGoalpostNeg(), field.friendlyGoalpostPos(),
-                              ball.position(), radius);
-
-            // restrict the goalie to a semicircle inscribed inside the defense area
-            Point goalie_restricted_pos =
-                field.friendlyGoal() - (field.friendlyDefenseArea().yLength() *
-                                        (field.friendlyGoal() - goalie_pos).norm());
-
-            // restrict the point to be within the defense area
-            auto goalie_orientation = (ball.position() - goalie_pos).orientation();
-            next_intent             = move_action.updateStateAndGetNextIntent(
-                *robot, goalie_restricted_pos, goalie_orientation, 0.0, false, false,
+            // compute angle between two vectors, negative goal post to ball and positive
+            // goal post to ball
+            Angle block_cone_angle =
+                (ball.position() - field.friendlyGoalpostNeg())
+                    .orientation()
+                    .minDiff(
+                        (ball.position() - field.friendlyGoalpostPos()).orientation());
+
+            // compute block cone position, allowing 1 ROBOT_MAX_RADIUS_METERS extra on
+            // either side
+            Point goalie_pos = calcBlockCone(
+                field.friendlyGoalpostNeg(), field.friendlyGoalpostPos(), ball.position(),
+                block_cone_radius * block_cone_angle.toRadians());
```

Comment:
We mention that 'block_cone_radius' is related somewhat to ROBOT_MAX_RADIUS, but the definition of the dynamic parameter doesn't include ROBOT_MAX_RADIUS. Would it be possible to change that?

Reply:
updated ai.yaml